src/arch/riscv/isa.cc

```diff
@@ -575,10 +576,6 @@ ISA::setMiscReg(int misc_reg, RegVal val)
         auto hcounter = readMiscRegNoEffect(MISCREG_HCOUNTEREN);
         RegVal write_val = ((hcounter & ~(NEMU_COUNTER_MASK)) | (val & NEMU_COUNTER_MASK));
         setMiscRegNoEffect(MISCREG_HCOUNTEREN, write_val);
-    } else if (misc_reg == MISCREG_SCOUNTEREN) {
-        auto scounter = readMiscRegNoEffect(MISCREG_SCOUNTEREN);
-        RegVal write_val = ((scounter & ~(NEMU_COUNTER_MASK)) | (val & NEMU_COUNTER_MASK));
```

Comment:
I suggest not deleting this logic, but changing the `NEMU_COUNTER_MASK` definition to `0xffffffffULL` ?
Also, I think {M,S,H}COUNTEREN can be moved into the switch-case below because its new value does not involves cpu.v or other information.

Reply:
Yes, I think the suggestions are very reasonable.